### ISSUE TITLE
svelte/no-unused-props misinterprets props as internal properties

### ISSUE THREAD

**Issue by InkedCat** [sveltejs/eslint-plugin-svelte#1335]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.12.2

### What did you do?
I created a Svelte layout file, defined the props return type, but intentionally left one of the properties unused. The `svelte/no-unused-props rule` was never triggered on Windows. However, on Linux and macOS, ESLint correctly reported the unused prop as an error when linting the file.

<details>
<summary>Configuration</summary>

```
import prettier from 'eslint-config-prettier';
import { includeIgnoreFile } from '@eslint/compat';
import js from '@eslint/js';
import svelte from 'eslint-plugin-svelte';
import globals from 'globals';
import { fileURLToPath } from 'node:url';
import ts from 'typescript-eslint';
import svelteConfig from './svelte.config.js';

const gitignorePath = fileURLToPath(new URL('./.gitignore', import.meta.url));

export default ts.config(
	includeIgnoreFile(gitignorePath),
	js.configs.recommended,
	...ts.configs.recommended,
	...svelte.configs.recommended,
	prettier,
	...svelte.configs.prettier,
	{
		languageOptions: {
			globals: { ...globals.browser, ...globals.node }
		},
		rules: {
			// typescript-eslint strongly recommend that you do not use the no-undef lint rule on TypeScript projects.
			// see: https://typescript-eslint.io/troubleshooting/faqs/eslint/#i-get-errors-from-the-no-undef-rule-about-global-variables-not-being-defined-even-though-there-are-no-typescript-errors
			'no-undef': 'off'
		}
	},
	{
		files: ['**/*.svelte', '**/*.svelte.ts', '**/*.svelte.js'],
		languageOptions: {
			parserOptions: {
				projectService: true,
				extraFileExtensions: ['.svelte'],
				parser: ts.parser,
				svelteConfig
			}
		}
	}
);
```
</details>

```svelte
<script lang="ts">
	import type { Snippet } from 'svelte';
	import type { LayoutData } from './$types';

	let { children }: { data: LayoutData; children: Snippet } = $props();
</script>

{@render children?.()}
```

### What did you expect to happen?

I expected this snippet to trigger the rule, since the `data` field is declared in the type but not used.

### What actually happened?

The rule is never applied, ESLint does not report any errors. This was tested on GitHub Runners and locally, and the issue only occurs in Windows environments. Linux and macOS are unaffected.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://github.com/InkedCat/no-unused-props-bug-env

Here is a link to one of my CI run where the bug happened: https://github.com/InkedCat/Vault-Launcher/actions/runs/17535461642

### Additional comments

I've done some debugging and believe I've found the source of the bug. In the `checkUnusedProperties` function (line 240 of no-unused-props.ts), the `for (const prop of properties)` loop checks whether properties are imported types or from internal sources. The `isInternalProperty` function in the same file compares the source file path inferred by Typescript declarations with the one passed down by ESLint. 

The issue arises from how file paths are handled on Windows: ESLint does normalize the paths (they contains double back slashes, `\\`), whereas Typescript does not and gets Windows like paths. Therefore the property is considered internal because the paths are differents.

Path format example:
Eslint: `C:\\Users\\...\\VaultLauncher\\src\\lib\\components\\navbar\\nav-main.svelte`
Typescript: `C:/Users/.../VaultLauncher/src/lib/components/navbar/nav-main.svelte`

After applying a small local patch to normalize the Typescript path, the rule is now correctly triggered on Windows.
```Typescript
		function isInternalProperty(symbol: ts.Symbol): boolean {
			const declarations = symbol.getDeclarations();
			if (!declarations || declarations.length === 0) return false;

			return declarations.every(
				(decl) => path.normalize(decl.getSourceFile().fileName) === fileName
			);
		}
```

If needed, I can take some time to open a PR and fix this 👍

**Comment by baseballyama** [comment]
Thank you for your investigation! If you’d like to open a PR, please feel free to do so. I can also submit one myself.

**Comment by InkedCat** [comment]
I would be happy to open a PR but I'm not sure about adding tests for this bug. The CI is only running on Ubuntu as of now, would it still be relevant ?

**Comment by baseballyama** [comment]
Previously I've created a PR for it but not competed. I think I need to finish the PR😅
https://github.com/sveltejs/eslint-plugin-svelte/pull/1122

**Comment by InkedCat** [comment]
I've tried to do it too and It seems there's errors similar to this one, there's even an additional disk letter `D:` before the path passed to `tools/write.ts`. Some commands needs to be adapted to Powershell too. https://github.com/InkedCat/eslint-plugin-svelte/actions/runs/17581306903/job/49938101495

**Comment by baseballyama** [comment]
Since I don’t currently have a Windows machine, I’d really appreciate it if you could make the fix and submit a PR.

**Comment by InkedCat** [comment]
No problem, I'll try to take some time to work on this two issues !

**Comment by InkedCat** [comment]
I've made a [PR ](https://github.com/sveltejs/eslint-plugin-svelte/pull/1355)to solve this issue and added support for Windows environments in the process
